src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Tree/Dependencies/Subscriptions/ProjectRuleHandler.cs

```diff
@@ -114,7 +115,7 @@ private void OnOtherProjectDependenciesChanged(IDependenciesSnapshot otherProjec
             var dependencyThatNeedChange = new List<IDependency>();
             foreach(var target in projectSnapshot.Targets)
             {
-                foreach (var dependency in target.Value.TopLevelDependencies)
+                foreach (var dependency in target.Value.TopLevelDependencies.Where(d => StringComparers.DependencyProviderTypes.Equals(d.ProviderType, ProviderTypeString)))
```

Comment:
It would have it was an array.  My question should have been:  is the underlying type index-able.  Since its not this looks fine.